### PR TITLE
Add python-libarchive-c

### PR DESCRIPTION
**Pull request by bollwyvl**
This adds https://github.com/Changaco/python-libarchive-c

As suggested over on https://github.com/conda-forge/libarchive-feedstock/pull/11, this includes an activate script to help it find the `libarchive` from there.

### CODE REVIEW

File: recipes/libarchive-c/activate.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+if [ "`uname`" == 'Darwin' ]
+then
+    # for Mac OSX
+    export LIBARCHIVE="${PREFIX}/lib/libarchive.dylib"
```
Comment (isuruf) [inline recipes/libarchive-c/activate.sh]: This has to be `CONDA_PREFIX` instead of `PREFIX`

File: recipes/libarchive-c/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
+{% set name = "libarchive-c" %}
```
Comment (isuruf) [inline recipes/libarchive-c/meta.yaml]: I'm not sure whether this is the best name. Sounds like a C library to me.
`python-libarchive-c` is the name of the repo, how about using that?

File: recipes/libarchive-c/activate.sh (hunk 1 above)
Comment (bollwyvl) [inline recipes/libarchive-c/activate.sh]: Good catch!

File: recipes/libarchive-c/meta.yaml (hunk 2 above)
Comment (bollwyvl) [inline recipes/libarchive-c/meta.yaml]: The pypi upstream has conditioned me, but totally right in this case.

**Comment by bollwyvl** [comment]
Made the suggested (and seconded!) changes.

Tested this out locally on OSX and linux, so am hopeful re CI.

Also read about [reprotest](https://pypi.org/project/reprotest/) on this [update from the debian folks](https://lists.debian.org/debian-devel-announce/2017/07/msg00004.html) and have started looking into what it would take to adapt for use in a conda(-forge) setting...

**Comment by bollwyvl** [comment]
Hooray!

File: recipes/python-libarchive-c/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+build:
+  number: 0
+  skip: true  # [win]
+
+requirements:
+  build:
+    - libarchive
+    - python
+    - setuptools
+    - toolchain
```
Comment (isuruf) [inline recipes/python-libarchive-c/meta.yaml]: I don't think you need `toolchain` or `libarchive` here.

Comment (bollwyvl) [inline recipes/python-libarchive-c/meta.yaml]: Indeed! Thanks for the speedy review!

File: recipes/python-libarchive-c/build.sh [hunk 4]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+export DETERMINISTIC_BUILD=1
```
Comment (bollwyvl) [inline recipes/python-libarchive-c/build.sh]: Hey, gotta start somewhere.

Comment (jakirkham) [inline recipes/python-libarchive-c/build.sh]: What does this do?

Comment (bollwyvl) [inline recipes/python-libarchive-c/build.sh]: Absolutely nothing... as it depends on some nix-specific patches to Python. Removed.

**Comment by synapticarbors** [comment]
It seems like a bunch of PR depend on this one. Is there anything blocking this one from being merged?

**Comment by bollwyvl** [comment]
Huzzah, thanks @isuruf! @synapticarbors I'll work on tidying up the others once there are builds!

**Pull request merged**